pylint/reporters/text.py

```diff
@@ -189,7 +194,13 @@ def on_set_current_module(self, module: str, filepath: Optional[str]) -> None:
 
     def write_message(self, msg: Message) -> None:
         """Convenience method to write a formatted message with class default template"""
-        self.writeln(msg.format(self._template))
+        self_dict = msg._asdict()
+        for key, value in self_dict.items():
+            # pylint: disable=fixme
+            # TODO: Add column to list of attributes to be printed as an empty string
+            if value is None and key in PRINT_AS_EMPTY_STRING:
+                self_dict[key] = ""
```

Comment:
Is a `for` - loop really necessary here? Maybe iterate only over `PRINT_AS_EMPTY_STRING` and do
```py
self_dict[key] = self_dict[key] or ""
```

Reply:
Or even better, what about this?

Initialize `end_lineno` and `end_col_offset` with an empty string instead `""`. You still would be able to differentiate  between a null value and a number. I.e. `int | Literal[""]` as type.

Reply:
> Or even better, what about this?
> 
> Initialize `end_lineno` and `end_col_offset` with an empty string instead `""`. You still would be able to differentiate between a null value and a number. I.e. `int | Literal[""]` as type.

I don't really like that as then we we're changing the type of `end_lineno` and `end_col_offset` while still "internal" to fit a specific type of reporter. For example, for the `JSONReporter` it makes no sense to have it be `""`. I'm not sure if other developer have written other `Reporter` but I think keeping both attributes `None` until a specific reporter starts handling them is the better option here.
I'll add your first suggestion!